Sources/SwifterSwift/SwiftStdlib/StringExtensions.swift

```diff
@@ -606,23 +606,45 @@ public extension String {
     ///		"Hello World!"[safe: 6..<11] -> "World"
     ///		"Hello World!"[safe: 21..<110] -> nil
     ///
-    /// - Parameter range: Half-open range.
-    subscript(safe range: CountableRange<Int>) -> String? {
-        guard let lowerIndex = index(startIndex, offsetBy: max(0, range.lowerBound), limitedBy: endIndex) else { return nil }
-        guard let upperIndex = index(lowerIndex, offsetBy: range.upperBound - range.lowerBound, limitedBy: endIndex) else { return nil }
-        return String(self[lowerIndex..<upperIndex])
+    /// - Parameter substringRange: Half-open range.
+    subscript(safe substringRange: CountableRange<Int>) -> String? {
+        guard
+            !isEmpty,
+            substringRange.lowerBound >= 0,
+            let lowerIndex = index(startIndex, offsetBy: substringRange.lowerBound, limitedBy: endIndex),
+            let upperIndex = index(lowerIndex, offsetBy: substringRange.upperBound - substringRange.lowerBound, limitedBy: endIndex),
+            let range = range(of: self) else {
+                return nil
+        }
+
+        if range ~= lowerIndex {
+            return String(self[lowerIndex..<upperIndex])
+        } else {
+            return nil
+        }
     }
 
     /// SwifterSwift: Safely subscript string within a closed range.
     ///
     ///		"Hello World!"[safe: 6...11] -> "World!"
     ///		"Hello World!"[safe: 21...110] -> nil
     ///
-    /// - Parameter range: Closed range.
-    subscript(safe range: ClosedRange<Int>) -> String? {
-        guard let lowerIndex = index(startIndex, offsetBy: max(0, range.lowerBound), limitedBy: endIndex) else { return nil }
-        guard let upperIndex = index(lowerIndex, offsetBy: range.upperBound - range.lowerBound, limitedBy: endIndex) else { return nil }
-        return String(self[lowerIndex...upperIndex])
+    /// - Parameter substringRange: Closed range.
+    subscript(safe substringRange: ClosedRange<Int>) -> String? {
+        guard
+            !isEmpty,
+            substringRange.lowerBound >= 0,
+            let lowerIndex = index(startIndex, offsetBy: substringRange.lowerBound, limitedBy: endIndex),
+            let upperIndex = index(lowerIndex, offsetBy: substringRange.upperBound - substringRange.lowerBound, limitedBy: endIndex),
+            let range = range(of: self) else {
+                return nil
+        }
+        
+        if range ~= lowerIndex && range ~= upperIndex {
```

Comment:
Again, no need to check this.

Reply:
If there's is no additional checker, this test case fails `XCTAssertNil(oneCharStr[safe: 1..<1])`. It is returning empty String instead of `nil`

Reply:
This shouldn't be a problem here right? this is the ClosedRange method (...) so ..< will not be the case :))

Reply:
I've updated both ClosedRange and CountableRange methods 😅